shell/platform/android/io/flutter/embedding/engine/systemchannels/BackGestureChannel.java

```diff
@@ -0,0 +1,127 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.systemchannels;
+
+import android.annotation.TargetApi;
+import android.window.BackEvent;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import io.flutter.Build.API_LEVELS;
+import io.flutter.Log;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.JSONMethodCodec;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A {@link MethodChannel} for communicating back gesture events to the Flutter framework.
+ *
+ * <p>The BackGestureChannel facilitates communication between the platform-specific Android back
+ * gesture handling code and the Flutter framework. It enables the dispatch of back gesture events
+ * such as start, progress, commit, and cancellation from the platform to the Flutter application.
+ */
+public class BackGestureChannel {
+  private static final String TAG = "BackGestureChannel";
+
+  @NonNull public final MethodChannel channel;
+
+  /**
+   * Constructs a BackGestureChannel.
+   *
+   * @param dartExecutor The DartExecutor used to establish communication with the Flutter
+   *     framework.
+   */
+  public BackGestureChannel(@NonNull DartExecutor dartExecutor) {
+    this.channel = new MethodChannel(dartExecutor, "flutter/backGesture", JSONMethodCodec.INSTANCE);
```

Comment:
@maRci002 I'm noticing that most platform channel names are all lowercase. Can you change this to "flutter/backgesture"?